### PR TITLE
Implement the dollar-variable-no-namespaced-assignment rule

### LINKED ISSUE #556: Rule idea: forbid assignment to namespaced variables
When using a Sass module with `@use`, exported variables are writable currently (except for built-in modules, so you cannot reassign `math.$pi` to a different value).
I started a discussion in https://github.com/sass/sass/issues/3171 about that, as I find that this makes modules harder to use (the whole point of modules was to make it easier to reason about files locally, but we can still be impacted by assignments done anywhere else).

I think it would make sense to have a linter rule forbidding assignment to namespaced variables (which does not cover the case of `@use "some-module" as *` but covers all other cases):

```scss
@use "some-module" as identifier;

identifer.$var: "should be rejected";
```

The idea would be to forbid variable assignments of the form `<identifier>.$<identifier>`

### PR DESCRIPTION
**Pull request by stof**
Closes #556

### CODE REVIEW

File: src/rules/dollar-variable-no-namespaced-assignment/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { utils } from "stylelint";
+import { namespace } from "../../utils";
+
+export const ruleName = namespace("dollar-variable-no-namespaced-assignment");
+
+export const messages = utils.ruleMessages(ruleName, {
+  expected: "Expected $ variable name to match specified pattern"
```
Comment (kristerkari) [inline src/rules/dollar-variable-no-namespaced-assignment/index.js:7]: Same thing in this rule as in the other one:

I went and checked from the stylelint repo, and most of the `-no-` rules seem to use the `rejected: "Unexpected ..."` message format, should this rule also use it?

Comment (stof) [reply src/rules/dollar-variable-no-namespaced-assignment/index.js]: I fixed it (and the whole message was wrong anyway due to a bad copy-paste)

**Review by kristerkari — APPROVED** [review]
Looks good! This can probably go to the next release (v4.1.0)

**Comment by kristerkari** [comment]
This was released in 4.1.0:
https://github.com/stylelint-scss/stylelint-scss/releases/tag/v4.1.0

Thank you @stof !

**Pull request merged**